src/packet/packet_structure.rs

```diff
@@ -1,28 +1,48 @@
-use crate::infrastructure::DeliveryMethod;
+use crate::packet::{DeliveryGuarantee, OrderingGuarantee};
 use std::net::SocketAddr;
 
 #[derive(Clone, PartialEq, Eq, Debug)]
-/// This is a user friendly packet containing the payload from the packet and the endpoint from where it came.
+/// This is a user friendly packet containing the payload, endpoint, and reliability guarantees.
+/// A packet could have reliability guarantees to specify how it should be delivered and processed.
+///
+/// | Reliability Type                 | Packet Drop | Packet Duplication | Packet Order  | Packet Fragmentation |Packet Delivery|
+/// | :-------------:                  | :-------------: | :-------------:    | :-------------:  | :-------------:    | :-------------:
+/// |       **Unreliable Unordered**   |       Yes       |       Yes          |      No          |      No             |       No
+/// |       **Reliable Unordered**     |       No        |      No            |      No          |      Yes             |       Yes
+/// |       **Reliable Ordered**       |       No        |      No            |      Ordered |      Yes             |       Yes
+/// |       **Sequenced**              |       Yes       |      No            |      Sequenced |      No |       No
+///
+/// You are able to send packets with any the above guarantees.
 pub struct Packet {
     /// the endpoint from where it came
     addr: SocketAddr,
     /// the raw payload of the packet
     payload: Box<[u8]>,
     /// defines on how the packet will be delivered.
-    delivery_method: DeliveryMethod,
+    delivery: DeliveryGuarantee,
+    /// defines on how the packet will be ordered.
+    ordering: OrderingGuarantee,
 }
 
 impl Packet {
```

Comment:
Looks like you're missing `reliable_sequenced` and subsequent tests

Reply:
I left this one away due to some troubles before, I revisited this guarantee and I think I could make it work.